### PR TITLE
Feature/il 2860

### PR DESCRIPTION
**Pull request by kris-ipor**

### CODE REVIEW

File: contracts/amm/AmmSwapsLens.sol [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+                fixedInterestRate: swap.fixedInterestRate,
+                payoff: swapValue,
+                openTimestamp: swap.openTimestamp,
+                endTimestamp: swap.endTimestamp,
+                liquidationDepositAmount: swap.liquidationDepositAmount,
+                state: swap.state
+            });
+    }
+
+    function owner() public view virtual returns (address) {
```
Comment (pete-ipor) [inline contracts/amm/AmmSwapsLens.sol:338]: This should be removed, i think ?

**Review by pete-ipor — CHANGES_REQUESTED** [review]

File: contracts/amm/AmmSwapsLens.sol [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+        address msgSender,
+        IporTypes.IporSwapMemory memory iporSwap,
+        int256 payoff,
+        uint256 closeTimestamp
+    ) internal view returns (uint256) {
+        if (iporSwap.state != uint256(AmmTypes.SwapState.ACTIVE)) {
+            return 1;
+        }
+
+        if (msgSender != owner()) {
```
[the remark this answers is not in the document]
Comment (mario-ipor) [reply contracts/amm/AmmSwapsLens.sol:145]: We need this.

File: contracts/amm/AmmSwapsLens.sol [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+        uint256 offset,
+        uint256 chunkSize
+    ) external view override returns (uint256 totalCount, IAmmSwapsLens.IporSwap[] memory swaps) {
+        AssetConfiguration memory assetConfiguration = _getAssetConfiguration(asset);
+        IMiltonStorage miltonStorage = IMiltonStorage(assetConfiguration.miltonStorage);
+        (uint256 count, uint256[] memory swapIds) = miltonStorage.getSwapReceiveFixedIds(msg.sender, offset, chunkSize);
+        return (count, _mapSwapsReceiveFixed(assetConfiguration, swapIds));
+    }
+
+    function getMySwaps(
```
Comment (mario-ipor) [inline contracts/amm/AmmSwapsLens.sol:247]: remove getMySwap - we should not use msg.sender inside lens method

File: contracts/amm/AmmSwapsLens.sol (hunk 1 above)
Comment (mario-ipor) [reply contracts/amm/AmmSwapsLens.sol:338]: here we should use library which read from slot where owner address is stored in router

File: contracts/amm/AmmSwapsLens.sol (hunk 3 above)
Comment (kris-ipor) [reply contracts/amm/AmmSwapsLens.sol]: removed

File: contracts/amm/AmmSwapsLens.sol (hunk 1 above)
Comment (kris-ipor) [reply contracts/amm/AmmSwapsLens.sol]: removed as agreed

File: contracts/amm/AmmSwapsLens.sol [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+
+    function _mapSwapsPayFixed(AssetConfiguration memory assetConfiguration, uint256[] memory swapIds)
+        internal
+        view
+        returns (IAmmSwapsLens.IporSwap[] memory swaps)
+    {
+        MiltonStorageTypes.IporSwapId[] memory swapIdsWithDirection = new MiltonStorageTypes.IporSwapId[](
+            swapIds.length
+        );
+        for (uint256 i = 0; i != swapIds.length; i++) {
```
[the remark this answers is not in the document]
Comment (kris-ipor) [reply contracts/amm/AmmSwapsLens.sol]: changed

File: contracts/amm/AmmSwapsLens.sol [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+
+    function getMySwaps(
+        address asset,
+        uint256 offset,
+        uint256 chunkSize
+    ) external view returns (uint256 totalCount, IAmmSwapsLens.IporSwap[] memory swaps) {
+        AssetConfiguration memory assetConfiguration = _getAssetConfiguration(asset);
+        IMiltonStorage miltonStorage = IMiltonStorage(assetConfiguration.miltonStorage);
+        (uint256 count, MiltonStorageTypes.IporSwapId[] memory swapIds) = miltonStorage.getSwapIds(
+            msg.sender,
```
[the remark this answers is not in the document]
Comment (kris-ipor) [reply contracts/amm/AmmSwapsLens.sol]: done

File: contracts/amm/AmmSwapsLens.sol [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+        AssetConfiguration memory assetConfiguration = _getAssetConfiguration(asset);
+        IMiltonStorage miltonStorage = IMiltonStorage(assetConfiguration.miltonStorage);
+        (uint256 count, uint256[] memory swapIds) = miltonStorage.getSwapReceiveFixedIds(msg.sender, offset, chunkSize);
+        return (count, _mapSwapsReceiveFixed(assetConfiguration, swapIds));
+    }
+
+    function getMySwaps(
+        address asset,
+        uint256 offset,
+        uint256 chunkSize
```
[the remark this answers is not in the document]
Comment (kris-ipor) [reply contracts/amm/AmmSwapsLens.sol:71]: method replaced by getSwaps with accout as param

File: contracts/amm/AmmSwapsLens.sol [hunk 7]
```diff
@@ -0,0 +1,391 @@
...
+                        }
+                    }
+                }
+            }
+        }
+
+        return 0;
+    }
+
+    function _isLiquidator(address liquidator) internal view returns (bool) {
```
[the remark this answers is not in the document]
Comment (kris-ipor) [reply contracts/amm/AmmSwapsLens.sol]: code removed

File: contracts/amm/AmmSwapsLens.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (kris-ipor) [reply contracts/amm/AmmSwapsLens.sol]: removed

File: contracts/amm/AmmSwapsLens.sol [hunk 8]
```diff
@@ -0,0 +1,391 @@
...
+        daiTimeBeforeMaturityAllowedToCloseSwapByBuyer = _daiConfiguration.timeBeforeMaturityAllowedToCloseSwapByBuyer;
+        daiTimeBeforeMaturityAllowedToCloseSwapByCommunity = _daiConfiguration
+            .timeBeforeMaturityAllowedToCloseSwapByCommunity;
+
+        iporOracle = _iporOracle;
+
+        liquidator0 = _liquidator0;
+    }
+
+    function getClosableStatusForPayFixedSwap(address asset, uint256 swapId)
```
[the remark this answers is not in the document]
Comment (kris-ipor) [reply contracts/amm/AmmSwapsLens.sol]: method removed

File: contracts/amm/AmmSwapsLens.sol [hunk 9]
```diff
@@ -0,0 +1,391 @@
...
+
+    IIporOracle public immutable iporOracle;
+
+    address public immutable liquidator0;
+
+    constructor(
+        AssetConfiguration memory _usdcConfiguration,
+        AssetConfiguration memory _usdtConfiguration,
+        AssetConfiguration memory _daiConfiguration,
+        IIporOracle _iporOracle,
```
[the remark this answers is not in the document]
Comment (kris-ipor) [reply contracts/amm/AmmSwapsLens.sol]: ok, renamed, _ used for internal immutables now only

**Review by mario-ipor — APPROVED** [review]

**Review by pete-ipor — APPROVED** [review]

**Pull request merged**